server/services/items.js

```diff
@@ -283,11 +283,47 @@ const deleteBatchItem = async (id) => {
   }
 };
 
+/* given that the basket state is not persisted between refreshes,
+   items which are created from the batchItem can get lost
+   this method is used for the task-scheduler which will periodically fetch 
+   these lost items, update their batchItem & delete the item accordingly. */
+const getLostItems = async () => {
+  var conditions = {
+    batchId: { $ne: null },
+    isTemplateBatchItem: false,
+    status: 'in-shop',
+    inBasket: true,
```

Comment:
Am I right in assuming that this is the combination of properties that marks the item as "lost" - e.g. it is both in the shop _and_ in the basket... And is it not possible for this to be the case in any other circumstance - e.g. when someone is legitimately shopping this item? (sorry for my ignorance on this!)

Reply:
Yup, exactly. I just double-checked - when an item is added to the basket it gains the `inBasket: true` whilst still retaining  `status: 'in-shop'`, as soon as it is checked-out, the item's status is switched to `shopped` and becomes associated with a given shopperId. 

There is a potential issue where a shopper has added a batch-item to their basket at the exact moment this scheduled logic runs, so it would technically fetch the item in their basket. I'd hope that if we set the scheduler to run at a odd hour e.g. 3am, it would minimise this risk. If it did happen, I'm not exactly sure what would happen, I guess if they tried to check-out, it would error on that item - something to look into for error handling. 

If you have any other ideas that'd be appreciated :)